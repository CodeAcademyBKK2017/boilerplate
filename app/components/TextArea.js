import React, {Component} from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

class InputText extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text>
          Content
        </Text>
        <TextInput style={styles.inputBox} multiline = {true}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
    justifyContent: 'center'
  },
  inputBox: {
    width: 250,
    height: 100,
    borderColor: 'gray',
    borderWidth: 1
  }
});

export default InputText;

