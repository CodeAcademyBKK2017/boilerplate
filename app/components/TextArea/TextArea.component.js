import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

export default class TextArea extends Component {
  render () {
    return (
      <View>
        <Text style={styles.instructions}>Content</Text>
        <TextInput 
          style={styles.textInput}
          multiline= {true}
          numberOfLines= {4}
          underlineColorAndroid='transparent'
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  },
  textInput: {
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    height: 100, 
    width: 180,
    borderColor: 'gray', 
    borderWidth: 1
  }
});