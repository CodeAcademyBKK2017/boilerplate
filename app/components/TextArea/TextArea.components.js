import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

class TextArea extends Component {
  render () {
    return (
      <View>
        <Text style={styles.welcome}>
                Please enter note here
        </Text>
        <TextInput
          style={styles.textin}
          // onChangeText={this.state.nans}
          multiline = {true}
          underlineColorAndroid = 'transparent'
          numberOfLines = {4}
          value = ''
        />

      </View>
    );
  }
}
const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  textin: {
    width: 250,
    height: 100,
    borderColor: 'gray',
    borderWidth: 1
  }
});
export default TextArea;