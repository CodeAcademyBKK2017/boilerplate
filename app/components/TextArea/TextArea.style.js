import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  },
  textInput: {
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    height: 100, 
    width: 180,
    borderColor: 'gray', 
    borderWidth: 1
  }
});
