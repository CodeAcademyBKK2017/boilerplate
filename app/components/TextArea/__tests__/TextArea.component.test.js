import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../TextArea.component';

describe('TextArea', () => {
    
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextArea />
    );
    expect(tree).toBeDefined();
  });

  it('TextArea: Should return TextArea', () => { // example to test class methods
    const tree = renderer.create(<TextArea />);
    expect(tree).toMatchSnapshot();
  });
});