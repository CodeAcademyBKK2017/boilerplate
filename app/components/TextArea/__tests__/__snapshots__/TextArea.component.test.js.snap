// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea TextArea: Should return TextArea 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Content
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={4}
    style={
      Object {
        "alignItems": "flex-start",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "flex-start",
        "width": 180,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
